### PR TITLE
Rewards Progress Bar & Other Elements

### PR DESCRIPTION
**Pull request by kennethlien**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Built a rough progress bar, started restyling the rewards. Left off here because I have a headache from my screen, but please feel free to review now anyway.

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

https://callstack.github.io/react-native-paper/
used to create progress bar

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

Still lots of styling to be done! Couldn't get two rewards on one row, so thats still left over. Also! Could not get the solid star from fontawesome.

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"
Simulator Screen Shot - iPhone 11 Pro Max - 2020-03-07 at 16 46 06

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: components/rewards/RewardsCard.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+} from '../../styled/rewards';
+import { FontAwesome5 } from '@expo/vector-icons';
+import Colors from '../../assets/Colors';
+
+class RewardCard extends React.Component {
+  render() {
+    return (
+      <Card>
+        <StarIcon>
+          <FontAwesome5 name="star" size={24} color={Colors.darkerGreen} />
```
Comment (wangannie) [inline components/rewards/RewardsCard.js]: To make this solid, add the word solid here as a prop.
`<FontAwesome5 name="star" size={24} color={Colors.darkerGreen} />`

Also, the icon color is primaryGreen.

File: styled/rewards.js [hunk 2]
```diff
@@ -15,6 +16,35 @@ export const ScrollViewContainer = styled.ScrollView.attrs(props => ({
...
+  border-radius: 9px;
+  padding: 10px;
+  margin: 2% 5%;
+  flex-direction: row;
+  flex: 1;
+  width: 40%;
+  background-color: #c9e3d9;
+`;
+
+export const StarIcon = styled.View`
```
Comment (wangannie) [inline styled/rewards.js]: Don't forget this is supposed to be a white circle.

File: components/rewards/RewardsCard.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        </StarIcon>
+        <ContentContainer>
+          <ContentText>$5 reward</ContentText>
+          <ContentText2>1000 points</ContentText2>
```
Comment (wangannie) [inline components/rewards/RewardsCard.js]: You don't need separate containers for these. Pull straight from BaseComponents.

File: components/rewards/RewardsHome.js [hunk 4]
```diff
@@ -1,71 +1,68 @@
...
+      <ScrollView style={{ marginTop: 45 }}>
+        <WaluigiContainer>
+          <Overline>REWARD PROGRESS</Overline>
+          <Text style={{ fontSize: 10 }}> </Text>
+          <Text
+            style={{
+              fontSize: 30
+            }}>
+            {parseInt(user.points) % 1000} / 1000
+          </Text>
```
Comment (wangannie) [inline components/rewards/RewardsHome.js]: Tommy created a base component style for this in a recent PR called Big Title (so update this when that gets merged).

File: styled/rewards.js [hunk 5]
```diff
@@ -15,6 +16,35 @@ export const ScrollViewContainer = styled.ScrollView.attrs(props => ({
...
 
+export const Card = styled.View`
+  box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.2);
+  border-radius: 9px;
+  padding: 10px;
+  margin: 2% 5%;
+  flex-direction: row;
+  flex: 1;
+  width: 40%;
+  background-color: #c9e3d9;
```
Comment (wangannie) [inline styled/rewards.js]: Use Colors.js colors -- lightestGreen

File: components/rewards/RewardsHome.js [hunk 6]
```diff
@@ -1,65 +1,68 @@
...
+            color={Colors.primaryGreen}
           />
-          <Text style={{ fontSize: 3 }}> </Text>
-          <Body>
-            Earn{' '}
-            {`${1000 -
-              (parseInt(user.points) %
-                1000)} points to unlock your next $5 reward`}{' '}
+          <Body style={{ marginBottom: 30 }}>
+            Earn {`${1000 - (parseInt(user.points) % 1000)}`} points to unlock
```
Comment (wangannie) [inline components/rewards/RewardsHome.js]: Hardcoded rewards -- add TODO to update later when Annie updates Airtable

File: components/rewards/RewardsHome.js [hunk 7]
```diff
@@ -1,65 +1,68 @@
...
+  if (N % 2 == 1) {
+    foo.push(1);
+  }
+  return foo;
+}
+// TODO @kennethlien to fix infinite scroll
 function RewardsHome({ transactions, user }) {
   return (
     <View>
       <ScrollView style={{ marginTop: 45 }}>
```
Comment (wangannie) [inline components/rewards/RewardsHome.js]: Should be 24 px

File: components/rewards/RewardsHome.js [hunk 8]
```diff
@@ -1,65 +1,68 @@
...
-        <WaluigiContainer>
-          <Overline>REWARD PROGRESS</Overline>
-          <Text style={{ fontSize: 10 }}> </Text>
-          <Text
-            style={{
-              fontSize: 30
-            }}>
+        <ProgressBarContainer>
+          <Overline style={{ marginBottom: 15 }}>REWARD PROGRESS</Overline>
+          <Title style={{ marginBottom: 15 }}>
```
Comment (wangannie) [inline components/rewards/RewardsHome.js]: Bottom margin should be 2

File: components/rewards/RewardsHome.js [hunk 9]
```diff
@@ -1,65 +1,68 @@
...
-            color={Colors.green800}
+            color={Colors.primaryGreen}
           />
-          <Text style={{ fontSize: 3 }}> </Text>
-          <Body>
-            Earn{' '}
-            {`${1000 -
-              (parseInt(user.points) %
-                1000)} points to unlock your next $5 reward`}{' '}
+          <Body style={{ marginBottom: 30 }}>
```
Comment (wangannie) [inline components/rewards/RewardsHome.js]: Bottom margin 28

File: components/rewards/RewardsHome.js [hunk 10]
```diff
@@ -1,65 +1,68 @@
...
-          </Text>
-          <Text style={{ fontSize: 3 }}> </Text>
+          </Title>
           <ProgressBar
-            style={{
-              flex: 0.8,
-              height: 20,
-              borderRadius: 20
-            }}
+            style={{ height: 20, borderRadius: 20, marginBottom: 15 }}
```
Comment (wangannie) [inline components/rewards/RewardsHome.js]: Margin Bottom 8

File: components/rewards/RewardsHome.js [hunk 11]
```diff
@@ -1,65 +1,68 @@
...
-            {`${1000 -
-              (parseInt(user.points) %
-                1000)} points to unlock your next $5 reward`}{' '}
+          <Body style={{ marginBottom: 30 }}>
+            Earn {`${1000 - (parseInt(user.points) % 1000)}`} points to unlock
+            your next $5 reward
           </Body>
-          <Body> </Body>
-          <Overline>
+          <Overline style={{ marginBottom: 5 }}>
```
Comment (wangannie) [inline components/rewards/RewardsHome.js]: Margin bottom 8

File: styled/rewards.js [hunk 12]
```diff
@@ -16,29 +17,42 @@ export const ScrollViewContainer = styled.ScrollView.attrs(props => ({
...
 
 export const StarIcon = styled.View`
+  align-items: center;
+  justify-content: center;
+  height: 40px;
+  width: 40px;
+  border-radius: 20px;
+  background-color: #fff;
+  flex-direction: row;
+  align-items: center;
```
Comment (wangannie) [inline styled/rewards.js]: Repeat of line 32!

File: components/rewards/RewardTab.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+          <FontAwesome5
+            name="star"
+            solid
+            size={24}
+            color={Colors.primaryGreen}
+          />
+        </StarIcon>
+        <RewardDescriptionContainer>
+          <Subhead color={Colors.darkerGreen}>$5 Reward</Subhead>
+          <Caption color={Colors.primaryGreen}>1000 points</Caption>
```
Comment (wangannie) [inline components/rewards/RewardTab.js]: Can you make this darkerGreen

File: components/rewards/RewardsHome.js [hunk 14]
```diff
@@ -1,65 +1,68 @@
...
+        {createPairs(Math.floor(parseInt(user.points) / 1000)).map(a => (
+          <RewardCardContainer>
+            <RewardTab></RewardTab>
+            <RewardTab></RewardTab>
+          </RewardCardContainer>
+        ))}
+        {createSingle(Math.floor(parseInt(user.points) / 1000)).map(a => (
+          <RewardCardContainer>
+            <RewardTab></RewardTab>
+          </RewardCardContainer>
```
Comment (wangannie) [inline components/rewards/RewardsHome.js]: Update this to use flex-wrap!!

File: components/rewards/RewardsHome.js [hunk 15]
```diff
@@ -1,65 +1,68 @@
...
     <View>
       <ScrollView style={{ marginTop: 45 }}>
-        <WaluigiContainer>
-          <Overline>REWARD PROGRESS</Overline>
-          <Text style={{ fontSize: 10 }}> </Text>
-          <Text
-            style={{
-              fontSize: 30
-            }}>
+        <ProgressBarContainer>
```
Comment (wangannie) [inline components/rewards/RewardsHome.js]: This naming is confusing -- it would seem like a container for the progress bar alone. Can you change this to RewardProgressContainer or something?

File: components/rewards/RewardsHome.js (hunk 14 above)
Comment (wangannie) [inline components/rewards/RewardsHome.js]: You can also move all of these rewards cards into the container above (currently named ProgressBarContainer) since you're applying the same left and right margins anyways.

File: components/rewards/RewardTab.js [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+import {
+  RewardTabContainer,
+  RewardDescriptionContainer,
+  StarIcon
+} from '../../styled/rewards';
+import { Subhead, Caption } from '../BaseComponents';
+import { FontAwesome5 } from '@expo/vector-icons';
+import Colors from '../../assets/Colors';
+
+class RewardTab extends React.Component {
```
Comment (wangannie) [inline components/rewards/RewardTab.js]: Can you rename to RewardCard? 'Tab' seems confusing because this is all in what we'd call rewards tab.

File: components/rewards/RewardTab.js [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+  render() {
+    return (
+      <RewardTabContainer>
```
Comment (wangannie) [inline components/rewards/RewardTab.js]: Rename to RewardCardContainer

Comment (wangannie) [reply components/rewards/RewardTab.js]: I know you used this name elsewhere, but you're probably getting rid of that once you change this all to flex right..?

File: components/rewards/RewardsHome.js (hunk 15 above)
Comment (wangannie) [reply components/rewards/RewardsHome.js]: Actually I just realized the Available Rewards overline is also in this. But also see my other comment about how you can also put the cards into here and rid of the extra 5% left/right margin
